homeassistant/components/sonos/media_player.py

```diff
@@ -759,32 +769,47 @@ def update_media_music(self, update_media_position, track_info):
         if playlist_position > 0:
             self._queue_position = playlist_position - 1
 
-    def update_volume(self, event=None):
+    @callback
+    def async_update_volume(self, event):
         """Update information about currently volume settings."""
-        if event:
-            variables = event.variables
+        variables = event.variables
 
-            if "volume" in variables:
-                self._player_volume = int(variables["volume"]["Master"])
+        if "volume" in variables:
+            self._player_volume = int(variables["volume"]["Master"])
 
-            if "mute" in variables:
-                self._player_muted = variables["mute"]["Master"] == "1"
+        if "mute" in variables:
+            self._player_muted = variables["mute"]["Master"] == "1"
 
-            if "night_mode" in variables:
-                self._night_sound = variables["night_mode"] == "1"
+        if "night_mode" in variables:
+            self._night_sound = variables["night_mode"] == "1"
 
-            if "dialog_level" in variables:
-                self._speech_enhance = variables["dialog_level"] == "1"
+        if "dialog_level" in variables:
+            self._speech_enhance = variables["dialog_level"] == "1"
 
-            self.schedule_update_ha_state()
-        else:
-            self._player_volume = self.soco.volume
-            self._player_muted = self.soco.mute
-            self._night_sound = self.soco.night_mode
-            self._speech_enhance = self.soco.dialog_mode
+        self.async_write_ha_state()
+
+    def update_volume(self):
+        """Update information about currently volume settings."""
+        self._player_volume = self.soco.volume
+        self._player_muted = self.soco.mute
+        self._night_sound = self.soco.night_mode
+        self._speech_enhance = self.soco.dialog_mode
 
     def update_groups(self, event=None):
         """Handle callback for topology change event."""
+        coro = self.async_update_groups_coro(event)
```

Comment:
If we prefix a function name with `async_` it should mean that we should call the function from inside the event loop.

Reply:
Maybe `create_update_groups_coro`?

Reply:
Sounds good.

Reply:
Resolved in https://github.com/home-assistant/core/pull/48684 (but docker hub is broken at the moment so the CI is likely to fail)